src/tcgwars/logic/impl/gen3/DragonFrontiers.groovy

```diff
@@ -204,44 +204,16 @@ public enum DragonFrontiers implements LogicCardInfo {
         weakness F
         pokeBody "Holon Veil", {
           text "Treat each Basic Pokémon and Evolution card in your deck, in your discard pile, in your hand, and in play as a Pokémon that has δ on its card."
-          onActivate {
-            bg.em().retrieveAndStore("holon_veil_counter_${self.owner}", {
-              if(it) {
-                it+1
-              } else {
-                def list = new CardList()
-                list += my.deck.filterByType(POKEMON)
-                list += my.discard.filterByType(POKEMON)
-                list += my.hand.filterByType(POKEMON)
-                list += my.prizeCardSet.filterByType(POKEMON)
-                list += my.lostZone.filterByType(POKEMON)
-                my.all.each {list += it.cards.filterByType(POKEMON)}
-                list = list.findAll{it.cardTypes.isNot(DELTA)}
-                list.each {it.cardTypes.add(DELTA)}
-                bg.em().storeObject("holon_veil_pokemon_${self.owner}", list)
-                1
-              }
-            })
-          }
-          onDeactivate {
-            bg.em().retrieveAndStore("holon_veil_counter_${self.owner}", {
-              if (it == 1) {
-                bg.em().retrieveAndStore("holon_veil_pokemon_${self.owner}", { CardList list ->
-                  list.each { it.cardTypes.remove(DELTA) }
-                  null
-                })
-                null
-              } else {
-                it - 1
-              }
-            })
+          getterA(GET_CARD_TYPES, BEFORE_LAST) {holder ->
+            if (holder.effect.target.player == self.owner)
```

Comment:
Add basic and evolution condition please

Reply:
> Add basic and evolution condition please

Cards prior to Diamond & Pearl that look for "Basic Pokémon or Evolution cards" are meant to search for any Pokémon (including, for example, LV.X which are neither)

> Q. Can I use Celio's Network to search my deck for a Pokémon LV.X?
> 
> A. Yes, you can. Earlier card references of "Basic Pokémon or Evolution card" refers to Pokémon in general, which includes Pokémon LV.X. (May 10, 2007 PUI Rules Team)

Actually need to go through older cards and correct that behaviour. The one card type that could be affected in all of this are Neo/e-Card Baby Pokémon, but even then those count as Basic Pokémon (and cards that don't allow babies explicitely exclude them).

Reply:
I mean this condition :) 

```suggestion
            if (holder.effect.target.player == self.owner && holder.object.is(POKEMON))
```

Reply:
Otherwise all cards (including energy and trainer) will be DELTA due to how this getter works.

Reply:
Oh. Sorry, didn't notice no check was being done my bad on that!